### DISCUSSION TITLE
Steren SHOME-117 3Gang Wall Switch

### DISCUSSION THREAD

**Discussion by carlitous** [RavenSystem/esp-homekit-devices#1772] [MEPLHAA Scripts Collection]
Device Name: SHOME-117
URL: https://www.steren.com.mx/apagador-wi-fi-triple.html
HAA version: 10.4.0
Summary of functions: 3 buttons, 1 per channel 110/240v 10A 600W max.

JSON: { "c":{"n":"Apagador-3S", "b":[ {"g":16, "t":5} ]}, "a":[{"0":{ "r":[ {"g":12} ]}, "1":{ "r":[ {"g":12, "v":1} ] }, "b":[ {"g":05} ], "s":5}, {"0":{ "r":[ {"g":13} ] }, "1":{ "r":[ {"g":13, "v":1} ] }, "b":[ {"g":16} ], "s":5}, {"0":{ "r":[ {"g":14} ] }, "1":{ "r":[ {"g":14, "v":1} ] }, "b":[ {"g":04} ], "s":5} ] }

Note: you can change the device name: "Apagador-3S" to your liking.
GPIO assignments were obtained from the following Tasmota template:
https://templates.blakadder.com/steren_SHOME-117.html

Long description: This is a TUYA app compatible 3 gang switch sold in Mexico under the Steren brand, it has it's own app though. Works with Alexa and Google straight out of the box. Aside from the rest this has physical push buttons instead of the most common touch type which makes it a bit more appealing to install.

Photos, additional information...:
To flash the device I followed the instructions from Tasmota GitHub:
https://tasmota.github.io/docs/Getting-Started/

FTDI 6 in 1 module used:
https://uelectronics.com/producto/modulo-convertidor-usb-6-en-1-ttl-a-rs232-rs485-cp2102-max485-max232/
DIP switch mode: 1:ON, 2:OFF.
Mode switch: 232-TTL

Tasmotizer Flashing tool:
https://github.com/tasmota/tasmotizer

To run it I followed Option 2 installation requirements using Terminal on a Macbook running macOS Monterey 12.2.1.

The following image shows the main board, very clear and easy to read and work with.
J1 (next to push button on the right) is the first pass-through junction for RX, TX, GND and V3.3 connections.
Next to the right of the R30 label is another pass-through junction for GPIO0. All of them are clearly labeled on the back of the board and much easier access.

![IMG_5516](https://user-images.githubusercontent.com/100984212/156868847-c47b60d7-7a52-4eed-b70e-f00c49856878.jpg)

Here it is connected to the FTDI adapter and fully working after flashing.
The loose wire (orange) is the one I used to short GPIO0 to GND momentarily.

![IMG_5529](https://user-images.githubusercontent.com/100984212/156869167-13d6d4e6-54aa-48a7-ace9-73597611823b.jpg)

**Comment by orson32** [comment]
tengo uno igual de steren pero de un solo botón, crees que me puedas ayudar con el JSON?

**Reply by carlitous** [reply]
Teoricamente este debería funcionar bien

```
{ "c":{"n":"Apagador-1S", "b":[ {"g":0, "t":5} ]},
  "a":[{
      "0":{ "r":[ {"g":12} ]},
      "1":{ "r":[ {"g":12, "v":1} ] },
      "b":[ {"g":0} ], "s":5
  }]
}
```
